### PR TITLE
Allow for export ONNX model with dict inputs

### PR DESCRIPTION
**Pull request by natuan**
This allows for passing in inputs of dict type to the export_onnx, enabling exported model with the original inp/out names.

### CODE REVIEW

File: src/sparseml/pytorch/utils/exporter.py [hunk 1]
```diff
@@ -151,6 +151,22 @@ def export_to_zoo(
             sample_originals=sample_originals,
         )
 
+    def get_output_names(self, out: Any):
```
Comment (bfineran) [inline src/sparseml/pytorch/utils/exporter.py:154]: nit: method should be static

**Review by bfineran — COMMENTED** [review]
LGTM overall, I'm worried that for models with multiple inputs, the proper input order will not be preserved by a `Dict` unless it is an `OrderedDict`. 

It might be safer to have `input_names` as a separate arg whose names should match the order of the sample input. Having a warning if the `Dict` is not ordered might also be ok though. If we keep `Dict` the transformers integration should be updated to use an ordered one.

File: src/sparseml/pytorch/utils/exporter.py (hunk 1 above)
Comment (natuan) [reply src/sparseml/pytorch/utils/exporter.py]: Fixed

**Comment by natuan** [comment]
> LGTM overall, I'm worried that for models with multiple inputs, the proper input order will not be preserved by a `Dict` unless it is an `OrderedDict`.
> 
> It might be safer to have `input_names` as a separate arg whose names should match the order of the sample input. Having a warning if the `Dict` is not ordered might also be ok though. If we keep `Dict` the transformers integration should be updated to use an ordered one.

Added the followings:

- Warning if sample batch passed in is a dict but not an ordered dict
- From transformers side, the sample batch constructed as OrderedDict

**Review by bfineran — DISMISSED** [review]
LGTM, thanks @natuan!

**Comment by bfineran** [comment]
looks like quality needs to run @natuan

**Review by bfineran — APPROVED** [review]

**Review by markurtz — APPROVED** [review]

**Pull request merged**